### ISSUE TITLE
When background is transparent, highlighted text not inverted

### ISSUE THREAD

**Issue by npezza93** [sourcelair/xterm.js#962]
In ZSH when you tab complete, you can tab through options. When you set xterms background to transparent(say you are on a dark background) the text color does not get inverted correctly on the v3 branch.  Changing the foreground, cursor, or selection colors did not fix the issue.

Repro steps: 
 - Set xterms theme background to transparent
 - jump into zsh
 - tab complete a command

If that doesn't really make sense I've attached a gif to help to show what I'm talking about. 
![xterm](https://user-images.githubusercontent.com/5059927/30245758-f7c6b600-95b1-11e7-970a-b96220fb6620.gif)

**Comment by Tyriar** [comment]
Transparent is no longer a valid background color. This is related to https://github.com/sourcelair/xterm.js/issues/936, previously black was just used as the default background color which breaks when the background is not black.

Did you call this to get into this state?

```js
var term = new Terminal({
  theme: {
    background: 'transparent'
  }
});
```

**Comment by npezza93** [comment]
@Tyriar Yes, that is what I called to get into that state.

**Comment by npezza93** [comment]
Changing the background to 'none', yields the desired results and inverts the text color correctly. Closing.

**Comment by corwin-of-amber** [comment]
With #1236, transparent background is allowed. But text highlighted with "inverse video" is still rendered incorrectly (invisible text over white background).

**Comment by corwin-of-amber** [comment]
I take that back – this was fixed in 4.3.0!
